src/core/runtime/dns-nodejs.js

```diff
@@ -2,10 +2,29 @@
 
 const dns = require('dns')
 const _ = require('lodash')
+const isIPFS = require('is-ipfs')
 const errcode = require('err-code')
 
+const maxRecursiveDepth = 32
+
 module.exports = (domain, opts, callback) => {
-  resolveDnslink(domain)
+  // recursive is true by default, it's set to false only if explicitly passed as argument in opts
+  const recursive = opts.recursive === undefined || opts.recursive.toString() === 'true'
```

Comment:
Default true for `null`/`undefined` and boolean value of argument otherwise:

```suggestion
  const recursive = opts.recursive == null ? true : Boolean(opts.recursive)
```

Reply:
`Boolean('false')` actually returns `true` not `false`. Arguments passed via query parameters are of `string` type and we need to parse those too. Once easy solution I just now came up with is this: 

`const recursive = opts.recursive.toString() !== 'false'`

If recursive is `false` or `"false"` then only we parse it as false. `true` when passed anything else, including `null` and `undefined`

Reply:
Edit: That wouldn't work either because error will be thrown if opts.recursive is undefined. 

The original one is probably the best solution after all, with slight modifications: 

`const recursive = opts.recursive == undefined || opts.recursive.toString() !== 'false'`

opts.recursive === undefined \|\| opts.recursive.toString() === 'true'
 





opts.recursive === undefined \|\| opts.recursive.toString() === 'true'
 

Reply:
> Arguments passed via query parameters are of string type and we need to parse those too

It should be passed as a boolean - this is the documented type for the value. If a string is being passed then we need to [fix our HTTP endpoint](https://github.com/ipfs/js-ipfs/blob/master/src/http/api/resources/dns.js) to properly parse query strings to the expected types for the calls to core.

We can't guarantee the passed value is a boolean but we _can_ ensure that the value is a boolean for our code by casting it.

Reply:
That makes sense. Right now I don't think query parameters are being parsed and "type-casted" before passing to the core API.  For this PR, I've included it here: 

https://github.com/ipfs/js-ipfs/blob/fb114f1ece896130ea5eddc59248c304b0ae6036/src/http/api/resources/dns.js#L14-L18